### ISSUE TITLE
Notice type should be mandatory

### ISSUE THREAD

**Issue by SamGuay** [osumontreal/docs.osumontreal.ca#25]
Lorsqu'on génère notre propre titre de notice, il n'y a aucune couleur différencier le titre du body de la notice. S'il pouvait y avoir un petit contraste, ce sera :fire:.

```go
{{% notice "Nice Looking Title" %}}
Plain shortcode
{{% /notice %}}
```

![image](https://user-images.githubusercontent.com/30598330/111920699-c74de400-8a66-11eb-9600-05e83952fcd0.png)

**Comment by dannycolin** [comment]
@SamGuay tu pourrais partager un snippet du code qui te donne ce résultat?

**Comment by SamGuay** [comment]
Oups, j'avais oublié de l'intégrer. Voir edit ci-haut. 

En regardant le code du shortcode: 
```html
<div class="notice-wrapper" {{ if len .Params | eq 2 }} id="{{ .Get 1 }}" {{ end }}>
  <p class="notice-title notice-{{ .Get 0 }}">
    <span class="notice-icon fa fa-exclamation-circle" aria-hidden="true"></span>
    {{ .Get 0 }}
  </p>
  <div class="notice-content">
    {{ .Inner }}
  </div>
</div>
```

On pourrait ajouter un param au shortcode pour qu'on puisse setter la couleur du notice indépendamment du titre, eg:
```html
<div class="notice-wrapper" {{ if len .Params | eq 3 }} id="{{ .Get 2 }}" {{ end }}>
  <p class="notice-title notice-{{ .Get 0 }}">
    <span class="notice-icon fa fa-exclamation-circle" aria-hidden="true"></span>
    {{ .Get 1 }}
  </p>
  <div class="notice-content">
    {{ .Inner }}
  </div>
</div>
``` 

Ce code est à améliorer, car si on met juste `warning`, `if len .Params` ne fonctionne plus si l'ordre est changé, etc. Bug prévu si qqu'un utilise seulement 2 params (eg, css class + id). On pourrait soit nommer tous les params (mais chiant quand on écrit du contenu) ou seulement nommer le id= si utilisé..

**Comment by dannycolin** [comment]
> Oups, j'avais oublié de l'intégrer. Voir edit ci-haut.

Ah! C'est que tu ne lui donne pas un "type" (warning, info). Est-ce que c'est réellement un bug dans ce cas-ci :p ?

> On pourrait ajouter un param au shortcode pour qu'on puisse setter la couleur du notice indépendamment du titre, eg:

Je pense qu'on devrait limiter ça à des couleurs prédéterminées, car:

1. On est sûr quelles ont un contraste suffissant 
2. Ça briserait l'association de la couleur à un type de notice (e.g rouge = attention!, bleu = info supplémentaire)

**Comment by SamGuay** [comment]
> Ah! C'est que tu ne lui donne pas un "type" (warning, info). Est-ce que c'est réellement un bug dans ce cas-ci :p ?

Oui, car tel que présenté dans l'exemple, le shortcode n'est pas _assez intelligent_ pour détecter si le param qu'on lui passe est un "type" ou un "titre" random. Par contre, ce bug pourrait devenir une feature et je crois qu'on converge vers la même idée :arrow_down: 

> Je pense qu'on devrait limiter ça à des couleurs prédéterminées, car...:

Je suis 100% d'accord avec toi. Quand je me relis j'avoue qu'on dirait que je veux dire "Je veux pouvoir mettre n'importe quelle couleur en passant une couleur en param". Ce que je voulais dire, c'était d'avoir 2 params différents, un pour le "type" (warning, info, tip, success, etc.) et un autre pour pouvoir spécifier le titre. 

```go
{{% notice info Title %}}
Saviez-vous que...
{{% /notice %}}
```
L'idéal serait que le shortcode détecte s'il n'y a pas de titre, de mettre le type en tant que titre. Comme i18n fonctionne aussi dans les shortcodes, la traduction ne serait pas un problème également.

**Comment by dannycolin** [comment]
> L'idéal serait que le shortcode détecte s'il n'y a pas de titre, de mettre le type en tant que titre. Comme i18n fonctionne aussi dans les shortcodes, la traduction ne serait pas un problème également.

C'est que les titres personnalisés ne sont pas supportés. J'ai changé cette issue pour régler le problème avec le *type*. J'en ouvre une autre pour ajouter l'option d'avoir un titre personnalisé.
